runtime/src/system_instruction_processor.rs

```diff
@@ -265,14 +268,15 @@ pub fn process_instruction(
     instruction_data: &[u8],
     invoke_context: &mut InvokeContext,
 ) -> Result<(), InstructionError> {
+    let transaction_context = &invoke_context.transaction_context;
+    let instruction_context = transaction_context.get_current_instruction_context()?;
```

Comment:
Can this ever fail?

Reply:
In production no, but it can in the tests, if one forgets to push an `InstructionContext`.